jinahub/rankers/LightGBMRanker/lightgbm_ranker.py

```diff
@@ -0,0 +1,209 @@
+from jina import Executor, DocumentArray, requests
+
+
+class LightGBMRanker(Executor):
+    """
+    Computes a relevance score for each match using a pretrained learning-to-rank model trained with LightGBM (https://lightgbm.readthedocs.io/en/latest/index.html)
+    :param query_features: name of the features to extract from query Documents and used to compute relevance scores by the model loaded
+    from model_path
+    :param match_features: name of the features to extract from match Documents and used to compute relevance scores by the model loaded
+    from model_path
+    :param relevance_label: If call :meth:`train` endpoint, the label will be used as groundtruth for model training. If on :meth:`rank` endpoint, the
+    label will be used to assign a score to :attr:`Document.scores` field.
+    :param model_path: path to the pretrained model previously trained using LightGBM.
+    :param params: Parameters used to train the LightGBM learning-to-rank model.
+    :param categorical_query_features: name of features contained in `query_features` corresponding to categorical features.
+    :param categorical_match_features: name of features contained in `match_features` corresponding to categorical features.
+    :param query_features_before: True if `query_features` must be placed before the `match` ones in the `dataset` used for prediction.
+    :param args: Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    .. note::
+        The name of the features are used to extract the features from incoming `documents`. Check how these features are accessed in
+        :class:`Document` at https://docs.jina.ai/api/jina.types.document/
+    """
+
+    def __init__(
+        self,
+        query_features: List[str] = ['query_price', 'query_size'],
+        match_features: List[str] = ['match_price', 'match_size'],
+        relevance_label: str = 'relevance',
+        model_path: Optional[str] = None,
+        params: Dict = {
+            'task': 'train',
+            'boosting_type': 'gbdt',
+            'objective': 'lambdarank',
+            'min_data_in_leaf': 1,
+            'feature_pre_filter': False,
+        },
+        categorical_query_features: Optional[List[str]] = None,
+        categorical_match_features: Optional[List[str]] = None,
+        query_features_before: bool = False,
+        *args,
+        **kwargs,
+    ):
+        super(LightGBMRanker, self).__init__(*args, **kwargs)
+        self.params = params
+        self.model_path = model_path
+        self.query_features = query_features
+        self.match_features = match_features
+        self.categorical_query_features = categorical_query_features
+        self.categorical_match_features = categorical_match_features
+        self.query_features_before = query_features_before
+        self.label = relevance_label
+        self.logger = JinaLogger(self.__class__.__name__)
+        if self.model_path and os.path.exists(self.model_path):
```

Comment:
How about we thrown an error, or at least a warning, if model path was provided but doesn't exist?